src/main/java/com/epam/reportportal/extension/azure/AzureExtension.java

```diff
@@ -240,39 +283,87 @@ public boolean testConnection(Integration integration) {
 
 	@Override
 	public Optional<Ticket> getTicket(String id, Integration integration) {
-		IntegrationParameters params = getParams(integration);
-		ApiClient defaultClient = getConfiguredApiClient(params.getPersonalAccessToken());
-		String organizationName = extractOrganizationNameFromUrl(defaultClient, params.getOrganizationUrl());
+		initFields(integration);
 
 		WorkItemsApi workItemsApi = new WorkItemsApi(defaultClient);
 		try {
 			WorkItem workItem = workItemsApi
 					.workItemsGetWorkItem(organizationName, Integer.valueOf(id), params.getProjectName(), API_VERSION,
 							null, null, null);
-			Ticket ticket = new Ticket();
-			ticket.setId(workItem.getId().toString());
-			ticket.setTicketUrl(workItem.getUrl());
-			ticket.setStatus(workItem.getFields().get("System.State").toString());
-			ticket.setSummary(workItem.getFields().get("System.Title").toString());
-			return Optional.of(ticket);
+			return Optional.of(convertWorkItemToTicket(workItem));
 		} catch (ApiException e) {
 			LOGGER.error("Unable to load ticket: " + e.getMessage(), e);
 			return Optional.empty();
 		}
 	}
 
 	@Override
-	// TODO: Implement in the future
+	// TODO: Add attachments from the test-item to the issue
 	public Ticket submitTicket(PostTicketRQ ticketRQ, Integration integration) {
-		return null;
+		initFields(integration);
+
+		List<JsonPatchOperation> patchOperationList = new ArrayList<>();
+
+		uploadAttachmentToAzure(ticketRQ);
+
+		String issueType = null;
+		List<PostFormField> fields = ticketRQ.getFields();
+		for (PostFormField field : fields) {
+			String id = replaceSeparators(field.getId());
```

Comment:
function is a bit long. please move "for" body code to a new function. of the whole part as getPatchOperationsForFields maybe?

Reply:
I extracted a method with "for" loop, because if I extract only a body then it is much more difficult to get the value of issueType.